### PR TITLE
Fixes #40 - Download urls and dependencies

### LINKED ISSUE #40: Download the pages with all their links to a certain depth
**Is your feature request related to a problem? Please describe.**
Currently only the requested page is being downloaded, but it is possible to download all the links it contains and so on with these new downloads

**Describe the solution you'd like**
so that the work system is not affected, the client can handle the issue of searching for new links within the downloads and then requesting them to the seeds

### PR DESCRIPTION
**Pull request by e1Ru1o**
Fixes #40 

**Changes:**
-   Receive data in bytes
-   Find links in html and request them to a seed
-   Create a folder for each url
-   Write dependencies in each

### CODE REVIEW

File: dispatcher.py [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
 from multiprocessing import Process, Queue
-from util.utils import parseLevel, makeUuid, LoggerFactory as Logger, noBlockREQ
+import zmq, time, os, mimetypes, pickle, json
+from util.utils import parseLevel, makeUuid, LoggerFactory as Logger, noBlockREQ, valid_tags, change_html
 
 
 log = Logger(name="Dispatcher")
+    
+    
+def writer(root, url, old, data, name, graph):
```
Comment (AntiD2ta) [inline dispatcher.py:17]: Summary for this pls

File: dispatcher.py [hunk 2]
```diff
@@ -1,25 +1,33 @@
-import zmq, time, re
-from util.params import urls
+from bs4 import BeautifulSoup
+
+from bs4 import BeautifulSoup
```
Comment (AntiD2ta) [inline dispatcher.py:2]: This line is repited

File: dispatcher.py [hunk 3]
```diff
@@ -222,6 +312,8 @@ def main(args):
     parser.add_argument('-a', '--address', type=str, default='127.0.0.1', help='node address')
     parser.add_argument('-p', '--port', type=int, default=4142, help='connection port')
     parser.add_argument('-l', '--level', type=str, default='DEBUG', help='log level')
+    parser.add_argument('-d', '--depth', type=int, default=1, help='recursively downloads depth')
```
Comment (AntiD2ta) [inline dispatcher.py]: i understand the help, but is not clear, a  "number of depth desired of recursively downloads"
i dont know, people usually dont know what recursively means

**Review by AntiD2ta — CHANGES_REQUESTED** [review]

File: dispatcher.py (hunk 3 above)
Comment (e1Ru1o) [reply dispatcher.py]: It is the same, yours has the words `number` and `desired`, but it is still the same, it is also a project for people who are sure to have an idea of ​​what it means `recursively`. But if you think it necessary feel free to change it

**Review by AntiD2ta — APPROVED** [review]

**Pull request merged**